pyopencl/array.py

```diff
@@ -612,12 +612,9 @@ def __init__(
 
                 shape = (shape,)
 
-            size = 1
-            for dim in shape:
-                size *= dim
-                if dim < 0:
-                    raise ValueError(
-                        f"negative dimensions are not allowed: {shape}")
+            size = np.prod(shape)
```

Comment:
Could you leave a comment here about `uint64 * int`? I would not have suspected this, and I could easily see other folks working with this code falling into this trap.